fix: handle XLSX files in GET DIRECT {CONSTANTS,DATA,LOOKUPS}

chrispcampbell opened the pull request:
Fixes #399 

See issue for more details.

I basically consolidated the handling of the 3 different kinds of tabular file sources:
* named csv file
* named xls[x] file
* tagged xls[x] file

And now any of those 3 sources work with any of:
* `GET DIRECT CONSTANTS`
* `GET DIRECT DATA`
* `GET DIRECT LOOKUPS`

Plenty of test coverage on this one.

ToddFincannonEI reviewed (commented):
I'm glad to see this cleaned up. The only potential problem I saw was in the  handleExcelOrCsvFile function, where the file extension is checked against lower case "csv". This would fail if the file name was upper case. I checked path.resolve and confirmed it does not normalize case.

chrispcampbell commented:
> I'm glad to see this cleaned up. The only potential problem I saw was in the handleExcelOrCsvFile function, where the file extension is checked against lower case "csv". This would fail if the file name was upper case. I checked path.resolve and confirmed it does not normalize case.

Good catch.  I updated the fix so that it converts the path to lowercase before checking the extension.  Also renamed one of the test files to have an uppercase CSV extension to verify this.  Will merge shortly.
